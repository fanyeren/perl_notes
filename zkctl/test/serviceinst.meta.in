#!/bin/bash
source $(dirname $0)/common
TYPE=service-instance
VALUE=bs_dnews_2
NODE=hankjin.vicp.net

raw='"category":"auto","service_id":"bs_dnews_0","slavenode_id":"cq01-2011q4-setest3-11.vm.baidu.com","state":"NEW","trans_db_priority":"1","up_down_state":"down"'

function reset()
{
    ./zkctl -s service-instance list | while read si; do
        ./zkctl -s service-instance del-meta-attr --service-instance-id=$si --key=slavenode_id
    done
}
#准备环境
node_clear $TYPE
node_clear service
node_clear slavenode
./zkctl -s slavenode add --slavenode-id=$NODE --data={}
./zkctl -w service add --service-id=$VALUE --data={}
echo "add service result=$?"

#set meta all
ID=$(./zkctl -w $TYPE add --service-id=$VALUE --slavenode-id=$NODE --data='{'$raw'}')
echo "add service instance result=$?"
./zkctl -s $TYPE get-meta-all --$TYPE-id=$ID
echo "get meta all result=$?"
./zkctl -s slavenode list-service-instance --slavenode-id=$NODE
echo "get service instance result=$?"
./zkctl -w $TYPE del --$TYPE-id=$ID 2>&1
echo "del service instance result=$?"

#set null meta
ID=$(./zkctl -w $TYPE add --service-id=$VALUE --slavenode-id=$NODE --data={})
./zkctl -s $TYPE get-meta-all --$TYPE-id=$ID

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE add-meta-attr --$TYPE-id=$ID --key=${kv[0]} --value=${kv[1]} 2>&1
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -s $TYPE get-meta-attr --$TYPE-id=$ID --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE set-meta-attr --$TYPE-id=$ID --key=${kv[0]} --value='hjz'
    echo "Set meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -s $TYPE get-meta-attr --$TYPE-id=$ID --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -s $TYPE get-meta-all --$TYPE-id=$ID
echo "got meta all $TYPE result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE del-meta-attr --$TYPE-id=$ID --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -s $TYPE get-meta-all --$TYPE-id=$ID
echo "got empty $TYPE result result=$?"
./zkctl -w $TYPE add-meta-attr --$TYPE-id=$ID --key=slavenode_id --value=$NODE

node_clear $TYPE
node_clear service
node_clear slavenode
