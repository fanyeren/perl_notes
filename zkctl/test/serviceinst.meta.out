add service result=0
add service instance result=0
{"category":"auto","service_id":"bs_dnews_0","slavenode_id":"hankjin.vicp.net","state":"NEW","trans_db_priority":"1","up_down_state":"down"}
get meta all result=0
bs_dnews_2-0000000001
get service instance result=0
del service instance result=0
{"slavenode_id":"hankjin.vicp.net"}
Add meta category=auto result 0
auto
Get meta category=auto result 0
Set meta category=auto result 0
hjz
Get meta category=auto result 0
Add meta service_id=bs_dnews_0 result 0
bs_dnews_0
Get meta service_id=bs_dnews_0 result 0
Set meta service_id=bs_dnews_0 result 0
hjz
Get meta service_id=bs_dnews_0 result 0
service-instance add-meta-attr failed, this key field may be exist already
Add meta slavenode_id=cq01-2011q4-setest3-11.vm.baidu.com result 1
hankjin.vicp.net
Get meta slavenode_id=cq01-2011q4-setest3-11.vm.baidu.com result 0
Set meta slavenode_id=cq01-2011q4-setest3-11.vm.baidu.com result 0
hjz
Get meta slavenode_id=cq01-2011q4-setest3-11.vm.baidu.com result 0
Add meta state=NEW result 0
NEW
Get meta state=NEW result 0
Set meta state=NEW result 0
hjz
Get meta state=NEW result 0
Add meta trans_db_priority=1 result 0
1
Get meta trans_db_priority=1 result 0
Set meta trans_db_priority=1 result 0
hjz
Get meta trans_db_priority=1 result 0
Add meta up_down_state=down result 0
down
Get meta up_down_state=down result 0
Set meta up_down_state=down result 0
hjz
Get meta up_down_state=down result 0
{"category":"hjz","service_id":"hjz","slavenode_id":"hjz","state":"hjz","trans_db_priority":"hjz","up_down_state":"hjz"}
got meta all service-instance result result=0
del meta attr category result=0
del meta attr service_id result=0
del meta attr slavenode_id result=0
del meta attr state result=0
del meta attr trans_db_priority result=0
del meta attr up_down_state result=0
{}
got empty service-instance result result=0
